src/Recoil.rei

```diff
@@ -15,29 +15,69 @@ type readWrite('value) = t('value, readWriteMode);
 [@bs.module "recoil"] external isRecoilValue: 'any => bool = "isRecoilValue";
 
 // Atom creation
+type trigger = [ | `get | `set];
+
+type atomEffect('value) = {
+  node: readWrite('value),
+  trigger,
+  setSelf: 'value => 'value,
+  resetSelf: unit => unit,
+  onSet: (~newValue: 'value, ~oldValue: 'value) => unit,
+};
+
 type atomConfig('value) = {
   key: string,
   default: 'value,
 };
 
+type atomWithEffectsConfig('value) = {
+  key: string,
+  default: 'value,
+  effects_UNSTABLE:
+    option(array(atomEffect('value) => option(unit => unit))),
```

Comment:
why `option`? can't it just be an empty array?

Reply:
Oh yeah, WIP. I initially started adding the config to the regular `atomConfig` before realising that'd break retrocompatibility (which wouldn't be ideal for an experimental feature 😄)